### DISCUSSION TITLE
Is gl_DepthRange supported? If so, how can we enable its use in our shaders?

### DISCUSSION THREAD

**Discussion by nolankramer** [vsg-dev/VulkanSceneGraph#1315] [Q&A] [answered]
I've tried using a shader with `gl_DepthRange`, however glslang spits out:
```
ERROR: 0:38: 'gl_DepthRange' : undeclared identifier
```

I did make sure that the GraphicsPipeline had it's `DepthStencilState::depthTestEnable = true`.

**Comment by robertosfield** [comment, marked as answer]
It doesn't look like Vulkan supports gl_DepthRange:

[GL_KHR_vulkan_glsl.txt](https://github.com/KhronosGroup/GLSL/blob/main/extensions/khr/GL_KHR_vulkan_glsl.txt )

Bit of document that mentions gl_DepthRange:
~~~
Overview

    This is version 100 of the GL_KHR_vulkan_glsl extension.

    This extension modifies GLSL to be used as a high-level language for the
    Vulkan API.  GLSL is compiled down to SPIR-V, which the Vulkan API
    consumes.

    The following features are removed:
        * default uniforms (uniform variables not inside a uniform block),
          except for opaque types
        * atomic-counters (those based on atomic_uint)
        * subroutines
        * shared and packed block layouts
        * the already deprecated texturing functions (e.g., texture2D())
        * the already deprecated noise functions (e.g., noise1())
        * compatibility-mode-only features
        * gl_DepthRangeParameters and gl_NumSamples
        * gl_VertexID and gl_InstanceID
        * aliasing of vertex input components
          (see the Vulkan API specification)
        * Rectangle textures (sampler*Rect, image*Rect)
~~~

**Comment by robertosfield** [comment]
So the question then is, what are you trying to achieve and what you are using gl_DepthRange for.

**Reply by nolankramer** [reply]
Ah, I was going to use it in a fragment shader for some fade-out with distance techniques.

However I think I can do the calculation myself after passing some relevant information from the vertex shader.

**Reply by nolankramer** [reply]
Also, it looks like the depth range in Vulkan is always 0.0 to 1.0 (unless otherwise configured).
